### PR TITLE
Feature/46315 pcs inbound approve evidence note transfer

### PR DESCRIPTION
**Pull request by GuidoWorkDev**
46315 inbound - approval process with unit tests fixes

### CODE REVIEW

File: src/EA.Weee.RequestHandlers/AatfEvidence/GetTransferEvidenceNoteForSchemeRequestHandler.cs [hunk 1]
```diff
@@ -35,12 +34,17 @@ public async Task<TransferEvidenceNoteData> HandleAsync(GetTransferEvidenceNoteF
...
+
             Condition.Requires(transferredScheme).IsNotNull();
 
-            var allowedAccess = authorization.CheckOrganisationAccess(evidenceNote.OrganisationId) || 
-                          authorization.CheckSchemeAccess(evidenceNote.RecipientId);
+            var allowedAccess = authorization.CheckOrganisationAccess(evidenceNote.OrganisationId) || authorization.CheckSchemeAccess(evidenceNote.RecipientId);
+
+            if (!allowedAccess)
+            {
+                throw new SecurityException($"The user does not have access to the organisation or scheme with note ID {request.EvidenceNoteId}");
```
Comment (NateHogan) [inline src/EA.Weee.RequestHandlers/AatfEvidence/GetTransferEvidenceNoteForSchemeRequestHandler.cs:46]: Why has this been duplicated?

File: src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/ReviewTransferNoteViewModelMap.cs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ﻿namespace EA.Weee.Web.Areas.Scheme.Mappings.ToViewModels
 {
+    using EA.Weee.Web.ViewModels.Shared;
```
Comment (NateHogan) [inline src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/ReviewTransferNoteViewModelMap.cs:3]: Unused Import

File: src/EA.Weee.Web/Areas/Scheme/ViewModels/ReviewTransferNoteViewModel.cs [hunk 3]
```diff
@@ -42,16 +42,17 @@ public NoteStatus SelectedEnumValue
             }
         }
 
-        public ReviewTransferNoteViewModel() : base(new List<string> { ApproveEvidenceNote, RejectEvidenceNote, ReturnEvidenceNote })
+        //public ReviewTransferNoteViewModel() : base(new List<string> { ApproveEvidenceNote, RejectEvidenceNote, ReturnEvidenceNote })
```
Comment (NateHogan) [inline src/EA.Weee.Web/Areas/Scheme/ViewModels/ReviewTransferNoteViewModel.cs:45]: Unnecessary comment

File: src/EA.Weee.Web/Areas/Scheme/Views/OutgoingTransfers/DownloadTransferNote.cshtml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+﻿@using EA.Weee.Core.Helpers
+@using EA.Weee.Web.Areas.Aatf.ViewModels
+@using EA.Weee.Web.Areas.Scheme.Controllers
+@using EA.Weee.Web.Areas.Scheme.ViewModels.ManageEvidenceNotes
+@using EA.Weee.Web.RazorHelpers
+@using MvcNavigationHelpers
+@*@model EA.Weee.Web.ViewModels.Shared.ViewTransferNoteViewModel*@
```
Comment (NateHogan) [inline src/EA.Weee.Web/Areas/Scheme/Views/OutgoingTransfers/DownloadTransferNote.cshtml:7]: Unnecessary comment

File: src/EA.Weee.RequestHandlers/AatfEvidence/GetTransferEvidenceNoteForSchemeRequestHandler.cs (hunk 1 above)
Comment (GuidoWorkDev) [reply src/EA.Weee.RequestHandlers/AatfEvidence/GetTransferEvidenceNoteForSchemeRequestHandler.cs:46]: good spot - this has been doubled-up recently during a manual merge that went through without being noticed.  duplicate is removed.

File: src/EA.Weee.Web/Areas/Scheme/Controllers/OutgoingTransfersController.cs [hunk 5]
```diff
@@ -91,11 +94,14 @@ public async Task<ActionResult> SubmittedTransfer(Guid pcsId, Guid evidenceNoteI
 
             using (var client = apiClient())
             {
-                var noteData = await client.SendAsync(User.GetAccessToken(),
-                    new GetTransferEvidenceNoteForSchemeRequest(evidenceNoteId));
+                var noteData = await client.SendAsync(User.GetAccessToken(), new GetTransferEvidenceNoteForSchemeRequest(evidenceNoteId));
 
-                var model = mapper.Map<ReviewTransferNoteViewModel>(new ViewTransferNoteViewModelMapTransfer(pcsId,
-                    noteData, null));
+                ReviewTransferNoteViewModel model = mapper.Map<ReviewTransferNoteViewModel>(new ViewTransferNoteViewModelMapTransfer(pcsId, noteData, null)
```
[the remark this answers is not in the document]
Comment (GuidoWorkDev) [reply src/EA.Weee.Web/Areas/Scheme/Controllers/OutgoingTransfersController.cs:99]: done !

File: src/EA.Weee.Web/Areas/Scheme/Controllers/TransferEvidenceController.cs [hunk 6]
```diff
@@ -11,6 +11,7 @@
     using EA.Prsd.Core.Mapper;
     using EA.Weee.Api.Client;
     using EA.Weee.Requests.Scheme;
+    using EA.Weee.Web.ViewModels.Shared;
```
[the remark this answers is not in the document]
Comment (GuidoWorkDev) [reply src/EA.Weee.Web/Areas/Scheme/Controllers/TransferEvidenceController.cs:14]: import is used on line 189 as I have made ViewTransferNoteViewModel shared.

File: src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/ReviewTransferNoteViewModelMap.cs (hunk 2 above)
Comment (GuidoWorkDev) [reply src/EA.Weee.Web/Areas/Scheme/Mappings/ToViewModels/ReviewTransferNoteViewModelMap.cs:3]: same as previous

File: src/EA.Weee.Web/Areas/Scheme/ViewModels/ReviewTransferNoteViewModel.cs (hunk 3 above)
Comment (GuidoWorkDev) [reply src/EA.Weee.Web/Areas/Scheme/ViewModels/ReviewTransferNoteViewModel.cs:45]: done !

File: src/EA.Weee.Web/Areas/Scheme/Views/OutgoingTransfers/DownloadTransferNote.cshtml (hunk 4 above)
Comment (GuidoWorkDev) [reply src/EA.Weee.Web/Areas/Scheme/Views/OutgoingTransfers/DownloadTransferNote.cshtml:7]: Done !

**Review by NateHogan — APPROVED** [review]

**Pull request merged**